sdk/core/Azure.Core.Experimental/tests/public/JsonDataObjectTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests.Public
+{
+    public class JsonDataObjectTests
+    {
+        #region Convert tests
+
+        [Test]
+        public void CannotConvertObjectToLeaf()
+        {
+            dynamic data = JsonDataTestHelpers.CreateFromJson(@"{ ""value"": 5 }");
+
+            // TODO: Standardize Exception types.
+            Assert.Throws<InvalidOperationException>(() => { var i = (int)data; });
+            Assert.Throws<InvalidOperationException>(() => { var b = (bool)data; });
+            Assert.Throws<InvalidOperationException>(() => { var s = (string)data; });
+            Assert.Throws<JsonException>(() => { var time = (DateTime)data; });
```

Comment:
We should provide a consistent exception experience to aid developers working with the type through the dynamic interface.